src/platform/mbed/ConnectivityManagerImpl.cpp

```diff
@@ -0,0 +1,473 @@
+}
+
+void ConnectivityManagerImpl::_OnPlatformEvent(const ChipDeviceEvent * event) {}
+
+CHIP_ERROR ConnectivityManagerImpl::ProvisionWiFiNetwork(const char * ssid, const char * key)
+{
+    _interface = WiFiInterface::get_default_instance();
+
+    ChipLogDetail(DeviceLayer, "connect to %s with %s password", ssid, key);
+    _interface->attach(
+        [](nsapi_event_t event, intptr_t data) { ChipLogDetail(DeviceLayer, "WiFi event: event = %d, data = %p", event, data); });
+
+    auto result = _interface->connect(ssid, key, NSAPI_SECURITY_WPA_WPA2);
+    ChipLogDetail(DeviceLayer, "Connection result: %d", result);
+    auto status = _interface->get_connection_status();
+    ChipLogDetail(DeviceLayer, "Connection status: %s", status2str(status));
+    mWiFiStationMode = kWiFiStationMode_Enabled;
+    if (status == NSAPI_STATUS_GLOBAL_UP)
+    {
+        mWiFiStationState = kWiFiStationState_Connected;
+        mIsProvisioned    = true;
+    }
+    else
+    {
+        mWiFiStationState = kWiFiStationState_NotConnected;
+    }
+
+    return CHIP_NO_ERROR;
+}
+
+void ConnectivityManagerImpl::_ClearWiFiStationProvision(void)
+{
+    _interface = WiFiInterface::get_default_instance();
+    _interface->set_credentials(NULL, NULL, NSAPI_SECURITY_NONE);
+    mWiFiStationMode  = kWiFiStationMode_Disabled;
+    mWiFiStationState = kWiFiStationState_NotConnected;
+    mIsProvisioned    = false;
+}
+int ConnectivityManagerImpl::ScanWiFi(int APlimit, ::chip::DeviceLayer::Internal::NetworkInfo * wifiInfo)
+{
+
+    _interface = WiFiInterface::get_default_instance();
+    if (!_interface)
+    {
+        printf("ERROR: No WiFiInterface found.\n");
+        return -1;
+    }
+    auto status = _interface->get_connection_status();
+    if (status != NSAPI_STATUS_GLOBAL_UP)
+    {
+        printf(" Currently device not connected to any WIFI  AP\n");
+    }
+    WiFiAccessPoint * ap;
+
+    int count = _interface->scan(NULL, 0);
+
+    if (count <= 0)
+    {
+        printf("scan() failed with return value: %d\n", count);
+        return 0;
+    }
+
+    count = count < APlimit ? count : APlimit;
+
+    ap    = new WiFiAccessPoint[count];
+    count = _interface->scan(ap, count);
+
+    if (count <= 0)
+    {
+        return 0;
+    }
+
+    for (int i = 0; i < count; i++)
+    {
+        sprintf(wifiInfo[i].WiFiSSID, "%s", ap[i].get_ssid());
+        wifiInfo[i].security = NsapiToNetworkSecurity(ap[i].get_security());
+        wifiInfo[i].BSSID[0] = ap[i].get_bssid()[0];
+        wifiInfo[i].BSSID[1] = ap[i].get_bssid()[1];
+        wifiInfo[i].BSSID[2] = ap[i].get_bssid()[2];
+        wifiInfo[i].BSSID[3] = ap[i].get_bssid()[3];
+        wifiInfo[i].BSSID[4] = ap[i].get_bssid()[4];
+        wifiInfo[i].BSSID[5] = ap[i].get_bssid()[5];
+        wifiInfo[i].RSSI     = ap[i].get_rssi();
+        wifiInfo[i].channel  = ap[i].get_channel();
+    }
+
+    delete[] ap;
+
+    return count;
+}
+
+CHIP_ERROR ConnectivityManagerImpl::OnStationConnected()
```

Comment:
I'm not seeing the event being posted. Can we capture it in the callback registered by `_interface->attach` ?

Reply:
To do